caravel/assets/visualizations/nvd3_vis.js

```diff
@@ -67,8 +67,23 @@ function nvd3Vis(slice) {
         return;
       }
 
+      const axisLabels = payload.data[0].values;
+      let stretchMargin = 0;
+      // Calculates the longest label size for stretching bottom margin
+      function adjustMargins() {
+        const pixelsPerCharX = 4.5; // approx, depends on font size
+        let maxLabelSize = 0;
+        for (let i = 0; i < axisLabels.length; i++) {
+          maxLabelSize = Math.max(axisLabels[i].x.length, maxLabelSize);
+        }
+        stretchMargin = Math.ceil(Math.max(stretchMargin, pixelsPerCharX * maxLabelSize));
+      }
+
       let width = slice.width();
       const fd = payload.form_data;
+
+      adjustMargins();
```

Comment:
with the above suggested changes this would become `const stretchMargin = calculateStretchMargin();`